Modules/TemperatureSensor/TemperatureSensor.cs

```diff
@@ -70,41 +73,53 @@ public void GenerateAnomaly(int value)
         }
         public override async Task<ExecutionResult> RunAsync()
         {
-            double frequency = 2.0;
-            int amplitute = 5;
-            int samplingRate = 100;
-            // Connect to Host
             HubConnection connection = await ConnectAsync("http://127.0.0.1:5000/visualizerhub");
-            int i = 0;
+
+            //begin simple generator
+            var comps = new WaveConfig[3];
+            comps[0] = new WaveConfig(WaveType.Sine, 0.0001, 70);
+            comps[1] = new WaveConfig(WaveType.Sine, 0.001, 30);
+            comps[2] = new WaveConfig(WaveType.Flat, 1, 1)
+            {
+                VerticalShift = 130
+            };
+            var g = new WaveGenerator.WaveGenerator(comps);
+
+            //intitialize FFT object, which encapsulates the whole business
+            FFT fft = new FFT(128, 10);
+
+            var valueCounter = 0;
             while (true)
             {
-                Temperature message = null;
-                
-                lock (_sync)
+                var newValue = g.Read();
```

Comment:
please name all variables in a way that a non-familiar with the project reader can understand the purpose&function of the variable